components/storage/blockdevice/COMPONENT_SDIO/SDIOBlockDevice.cpp

```diff
@@ -0,0 +1,397 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2019 ARM Limited
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+#include "sdio_api.h"
+#if DEVICE_SDIO_ASYNC
+#include "us_ticker_api.h"
+#endif
+#include "SDIOBlockDevice.h"
+#include "platform/mbed_debug.h"
+
+using namespace mbed;
+
+#if DEVICE_SDIO
+
+/*
+ *  defines
+ */
+
+#define SDIO_DBG 0       /*!< 1 - Enable debugging */
+#define SDIO_CMD_TRACE 0 /*!< 1 - Enable SD command tracing */
+
+#define SDIO_BLOCK_DEVICE_ERROR_WOULD_BLOCK -5001           /*!< operation would block */
+#define SDIO_BLOCK_DEVICE_ERROR_UNSUPPORTED -5002           /*!< unsupported operation */
+#define SDIO_BLOCK_DEVICE_ERROR_PARAMETER -5003             /*!< invalid parameter */
+#define SDIO_BLOCK_DEVICE_ERROR_NO_INIT -5004               /*!< uninitialized */
+#define SDIO_BLOCK_DEVICE_ERROR_NO_DEVICE -5005             /*!< device is missing or not connected */
+#define SDIO_BLOCK_DEVICE_ERROR_WRITE_PROTECTED -5006       /*!< write protected */
+#define SDIO_BLOCK_DEVICE_ERROR_UNUSABLE -5007              /*!< unusable card */
+#define SDIO_BLOCK_DEVICE_ERROR_NO_RESPONSE -5008           /*!< No response from device */
+#define SDIO_BLOCK_DEVICE_ERROR_CRC -5009                   /*!< CRC error */
+#define SDIO_BLOCK_DEVICE_ERROR_ERASE -5010                 /*!< Erase error: reset/sequence */
+#define SDIO_BLOCK_DEVICE_ERROR_WRITE -5011                 /*!< SPI Write error: !SPI_DATA_ACCEPTED */
+#define SDIO_BLOCK_DEVICE_ERROR_UNSUPPORTED_BLOCKSIZE -5012 /*!< unsupported blocksize, only 512 byte supported */
+#define SDIO_BLOCK_DEVICE_ERROR_READ_BLOCKS -5013           /*!< read data blocks from SD failed */
+#define SDIO_BLOCK_DEVICE_ERROR_WRITE_BLOCKS -5014          /*!< write data blocks to SD failed */
+#define SDIO_BLOCK_DEVICE_ERROR_ERASE_BLOCKS -5015          /*!< erase data blocks to SD failed */
+
+#define BLOCK_SIZE_HC 512 /*!< Block size supported for SD card is 512 bytes  */
+
+// Types
+#define SDCARD_NONE 0  /**< No card is present */
```

Comment:
same here as above, lets make them more debug friendly.  I would even do enum class as we are now C++1x